copier/main.py

```diff
@@ -146,6 +146,20 @@ class Worker:
     pretend: bool = False
     quiet: bool = False
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self._cleanup()
+
+    def _cleanup(self) -> None:
+        if self.template.is_temp_clone:
+            rmtree(
+                self.template.local_abspath,
+                ignore_errors=False,
+                onerror=handle_remove_readonly,
+            )
```

Comment:
I think the cleanup should exist in `Template`, which is the one that creates the directory. Then, here:

```suggestion
        self.template._cleanup()
```